tests/test_views.py

```diff
@@ -1,21 +1,87 @@
 """
-Tests for the frontend.
+Unit tests for the frontend code.
 """
 
+import os
+import tempfile
 import unittest
 
-import ga4gh.server as server
+import ga4gh.backends
 import ga4gh.cli
+import ga4gh.protocol as protocol
+import ga4gh.server as server
+
+
+class MockBackend(ga4gh.backends.Backend):
+    """
+    A mock Backend class for testing.
+    """
+    def __init__(self, dataDir):
+        self._dataDir = dataDir
+
+    def searchVariants(self, request):
+        response = protocol.GASearchVariantsResponse()
+        response.variants = []
+        return response
+
+    def searchVariantSets(self, request):
+        response = protocol.GASearchVariantSetsResponse()
+        response.variantSets = []
+        return response
+
 
 class TestFrontend(unittest.TestCase):
 
     def setUp(self):
+        server.app.config['TESTING'] = True
+        server.app.config['VariantBackend'] = MockBackend('/tmp/')
         self.app = server.app.test_client()
 
     def testServer(self):
         self.assertTrue('404' in self.app.get('/').data)
 
-    # TODO: Fill in actual test cases here...
+    def testRouteReferences(self):
+        for path in ['/references/1', 'references/1/bases', 'referencesets/1']:
+            self.assertTrue('404' in self.app.get(path).data)
+
+        for path in ['/referencesets/search', '/references/search']:
+            self.assertTrue('404' in self.app.post(path).data)
+
+    def testRouteCallsets(self):
+        self.assertTrue('404' in self.app.post('/callsets/search').data)
+
+    def testRouteReads(self):
+        for path in ['/reads/search', '/readgroupsets/search']:
+            self.assertTrue('404' in self.app.post(path).data)
+
+    def testRouteVariants(self):
+        for path in ['/variantsets/search', '/variants/search']:
+            # POST gets routed to input checking
+            self.assertTrue('Unsupported media type'
+                            in self.app.post(path).data)
+
+            self.assertTrue('Malformed request' in self.app.post(
+                path, headers={'Content-type': 'application/json'}).data)
+
+            # OPTIONS should return success
+            self.assertEqual(200, self.app.options(path).status_code)
+
+            # GET is not defined
+            self.assertTrue('405' in self.app.get(path).data)
```

Comment:
Why are we string matching against the body here and checking the status_code above? I think it would be better for all of these tests to be of the form `self.assertEqual(<HTTP_CODE>, self.app.options(path).status_code)`.